diff --git src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs
index 4c6d148c70..84ec69740d 100644
--- src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs
+++ src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs
@@ -78,7 +78,16 @@ namespace System.Net.Http
 
                         string spn = "HTTP/" + authUri.IdnHost;
                         ChannelBinding channelBinding = connection.TransportContext?.GetChannelBinding(ChannelBindingKind.Endpoint);
-                        NTAuthentication authContext = new NTAuthentication(isServer:false, challenge.SchemeName, challenge.Credential, spn, ContextFlagsPal.Connection, channelBinding);
+
+                        NTAuthentication ConstructNTAuth() =>
+                            new NTAuthentication(isServer: false, challenge.SchemeName, challenge.Credential, spn, ContextFlagsPal.Connection, channelBinding);
+
+                        NTAuthentication authContext =
+                            (request.Properties.TryGetValue("S4uIdentity", out object s4uIdentityObj) &&
+                            s4uIdentityObj is System.Security.Principal.WindowsIdentity s4uIdentity) ?
+                            System.Security.Principal.WindowsIdentity.RunImpersonated(s4uIdentity.AccessToken, () => ConstructNTAuth()) :
+                            ConstructNTAuth();
+
                         try
                         {
                             while (true)
